controllers/jenkins/pipeline/pipeline_metadata_controller_test.go

```diff
@@ -0,0 +1,356 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipeline
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/golang/mock/gomock"
+	"github.com/jenkins-zh/jenkins-client/pkg/core"
+	"github.com/jenkins-zh/jenkins-client/pkg/job"
+	"github.com/jenkins-zh/jenkins-client/pkg/mock/mhttp"
+	"github.com/jenkins-zh/jenkins-client/pkg/util"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubesphere.io/devops/pkg/api/devops/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+const (
+	organizationAPIPrefix = "/blue/rest/organizations"
+	organization          = "jenkins"
+)
+
+func (c *Reconciler) getPipelineAPI(pipelineName string, folders ...string) string {
+	api := fmt.Sprintf("%s/%s", organizationAPIPrefix, organization)
+	folders = append(folders, pipelineName)
+	for _, folder := range folders {
+		api = fmt.Sprintf("%s/pipelines/%s", api, folder)
+	}
+	return api
+}
+
+func (c *Reconciler) getPipelineBranchAPI(option *job.GetBranchesOption) string {
```

Comment:
These two functions do not relate to the struct `Reconciler`. How about making them a regular function.